src/app/features/tech-record/components/tech-record-change-type/tech-record-change-type.component.ts

```diff
@@ -35,33 +34,30 @@ export class ChangeVehicleTypeComponent {
     private router: Router,
     private store: Store<TechnicalRecordServiceState>,
     private technicalRecordService: TechnicalRecordService
-  ) {
-    this.globalErrorService.clearErrors();
-
-    this.technicalRecordService.selectedVehicleTechRecord$.pipe(take(1)).subscribe(vehicle => (this.vehicle = vehicle));
+  ) {}
 
-    this.technicalRecordService.viewableTechRecord$
+  ngOnInit(): void {
+    this.globalErrorService.clearErrors();
+    this.technicalRecordService.techRecord$
       .pipe(take(1))
       .subscribe(techRecord => (!techRecord ? this.navigateBack() : (this.techRecord = techRecord)));
   }
-
   get makeAndModel(): string {
     const c = this.techRecord;
```

Comment:
this should be renamed to something more appropriate

Reply:
How about we throw all this in the service